### DISCUSSION TITLE
padding、font、and  other question

### DISCUSSION THREAD

**Discussion by Xiaobaishushu25** [slint-ui/slint#2783] [Q&A] [answered]
Hello, I am a beginner using slint and I have some questions as follows:
①how to change the position of the control in HorizontalBox?I didn't find any attributes like translateY or bottom in alignment.i also  tried padding-top but no effect.what i want to do is put the address text at the bottom of the HorizontalBox.
this is my code:
```
component content {
    HorizontalBox { 
        Text {
            text:"address";
            padding-top: 30px;
        }
        path:=LineEdit { 
            placeholder-text: "githup";
        }
        name:=LineEdit { 
            placeholder-text: "name";
        }
        down:=iconButton{
            clicked => {
                connect.exec(path.text,name.text)
            }
        }
    }
}
```
![image](https://github.com/slint-ui/slint/assets/76581880/d4be23dd-eea1-4d97-a3d1-138665c6fe68)
②How to change the font on the window？this is my code and run without errors，but the window font has not changed at all.
```
import "./resource/font/FiraCode-Medium.ttf";
export component MainWindow inherits Window{
    default-font-family: "Fira Code Medium";
   // items
}
```
③i define a TextEdit and set the read-only is true，but  when I use the Chinese input method, I can actually input.and i also want to konw how to hide the blur border when the TextEdit is selected.
![image](https://github.com/slint-ui/slint/assets/76581880/ef27afa9-5808-4a99-bdb6-e74554e4723c)
Moreover, I would like to know if there are plans to launch a plugin for slint on Clion？
Thank you for your patient reading!

**Comment by hunger** [comment]
About CLion: [editors/README.md](https://github.com/slint-ui/slint/tree/master/editors#jetbrains-ide) has a section on how to set this up. We have no plans to write maintain a plugin ourselves at this time though.

**Comment by ogoffart** [comment, marked as answer]
1. You may want to use another layout. Or in this case, vertical-alignment property in the Text.

2. This looks correct font should change for every item inside the Window. Perhaps there is a bug or it didn't find the font for some reason. 
Unless you mean the title bar, then it is drawn by the window manager and the font can't be changed.

3. Looks like a bug.

**Reply by tronical** [reply]
And https://github.com/slint-ui/slint/pull/2821 fixes the third point :)
